lib/search/query_components/user_filter.rb

```diff
@@ -33,7 +33,11 @@ def filter_hash(filter)
 
       case filter.type
       when "string"
-        es_filters << terms_filter(field_name, values)
+        if filter.multivalue_query == :any
+          es_filters << terms_filter(field_name, values)
+        else
+          es_filters << bool_must_filter(field_name, values)
```

Comment:
It might be valuable to point out here that in the default/else case, results must satisfy all values, in contrast to any.